Open.Nat/Discovery/Searcher.cs

```diff
@@ -101,10 +101,18 @@ private void Receive(CancellationToken cancelationToken)
 
 				var localHost = ((IPEndPoint)client.Client.LocalEndPoint).Address;
 				var receivedFrom = new IPEndPoint(IPAddress.None, 0);
-				var buffer = client.Receive(ref receivedFrom);
-				var device = AnalyseReceivedResponse(localHost, buffer, receivedFrom);
+                try
+                {
+                    var buffer = client.Receive(ref receivedFrom);
+                    var device = AnalyseReceivedResponse(localHost, buffer, receivedFrom);
 
-				if (device != null) RaiseDeviceFound(device);
+                    if (device != null) RaiseDeviceFound(device);
+                }
+                catch(Exception e)
```

Comment:
This `try-catch` block should be placed in the `Search` `while` body. 

Also, the project uses Tabs, no spaces.